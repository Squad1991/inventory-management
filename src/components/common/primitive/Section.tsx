import { styled } from '@compiled/react';
import React from 'react';
import Stack from '~src/components/common/primitive/Stack';

interface SectionProps extends React.PropsWithChildren {
  heading: string;
}

const SectionStyles = styled(Stack)({
  gap: '1rem',
});

const Section: React.FC<SectionProps> = ({ heading, children }) => {
  return (
    <SectionStyles>
      <h2>{heading}</h2>
      {children}
    </SectionStyles>
  );
};

export default Section;
