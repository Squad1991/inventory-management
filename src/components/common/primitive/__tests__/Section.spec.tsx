import { render } from '@testing-library/react';
import React from 'react';
import Section from '~src/components/common/primitive/Section';

describe('Section', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Section heading="My header">
        <div>1</div>
        <div>2</div>
        <div>3</div>
      </Section>,
    );
    expect(container).toMatchSnapshot();
  });
});
