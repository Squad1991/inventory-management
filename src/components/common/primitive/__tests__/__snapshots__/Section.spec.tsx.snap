// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Section > should render correctly 1`] = `
<div>
  <style
    data-cmpld="true"
  >
    ._zulp1j6v{gap:1rem}
  </style>
  <style
    data-cmpld="true"
  >
    ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
  </style>
  <div
    class="_1e0c1txw _2lx21bp4 _zulp1j6v"
  >
    <h3>
      My header
    </h3>
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
  </div>
</div>
`;
